component.ts

```diff
@@ -0,0 +1,47 @@
+function fetchResults(query: string) {
+  return fetch(`API_DOMAIN/search?q=${query}`);
+}
```

Comment:
```suggestion
async function fetchResults(query: string) {
  const response = await fetch(`API_DOMAIN/search?q=${query}`);
  if (response.ok) {
  	return response.json() 
  }
  throw new Error(`${response.status} - Error while fetching users : `)
}
```

Important car fetch ne renvoit pas directement le JSON ! 

[Cf la doc](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch)